src/steps/device/converter.ts

```diff
@@ -39,6 +57,58 @@ export function createDeviceEntity(device: Device): Entity {
         agentVersion: device.agent_version,
         firstEnrollmentOn: parseTimePropertyValue(device.first_enrollment),
         lastEnrollmentOn: parseTimePropertyValue(device.last_enrollment),
+        'general.systemVersion': general.system_version,
```

Comment:
Is it ever possible that `general` (or any of the other properties) are undefined?

```suggestion
        'general.systemVersion': general?.system_version,
```

Looks like at some point, someone manually added a `DeviceDetails` type, and they really should have typed everything as optional (for safety).

```diff
export interface DeviceDetails {
- general: {
+ general?: {
-   device_id: string;
+   device_id?: string;
    ...
```